### PR TITLE
configure.ac: fix res_query detection

### PR DESCRIPTION
**Pull request by dx9**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -354,38 +354,45 @@ if test x"$linux" = x"yes"; then
...
+  LIBS="$2"
+  AS_CASE($1,
+          res_nquery, [resolve_func="res_nquery(0, 0, 0, 0, 0, 0)"],
+          res_query, [resolve_func="res_query(0, 0, 0, 0, 0)"],
+          AC_MSG_FAILURE([unknown resolv func]))
   AC_LINK_IFELSE([AC_LANG_PROGRAM([[
         #include <netinet/in.h>
         #include <resolv.h>
-      ]], [[res_nquery(0, 0, 0, 0, 0, 0);]])],
+      ]], [[$resolve_func;]])],
```
Comment (tchaikov) [inline configure.ac]: nit, could extract `(0, 0, 0, 0, 0, 0)` out and put it here.

Comment (tchaikov) [inline configure.ac]: nit, could extract `(0, 0, 0, 0, 0, 0)` out and put it here.

**Comment by dx9** [comment]
@tchaikov Ya much cleaner that way. Also added a comment and fixed the check order so res_nquery always takes precedence over res_query. Thanks

**Comment by dx9** [comment]
@tchaikov oops I broke it... Hang on

**Comment by dx9** [comment]
@tchaikov I think it's good now. LMK

File: configure.ac [hunk 2]
```diff
@@ -354,38 +354,42 @@ if test x"$linux" = x"yes"; then
...
-  AC_MSG_CHECKING([if res_nquery will link (LIBS=$1)])
+  AC_MSG_CHECKING([if $1 will link (LIBS=$3)])
   saved_LIBS="${LIBS}"
-  LIBS="$1"
+  LIBS="$3"
   AC_LINK_IFELSE([AC_LANG_PROGRAM([[
         #include <netinet/in.h>
         #include <resolv.h>
-      ]], [[res_nquery(0, 0, 0, 0, 0, 0);]])],
+      ]], [[$1$2;]])],
```
Comment (tchaikov) [inline configure.ac]: @dx9 does `"$1"(0, 0, 0, 0, 0, 0)` work here?

Comment (dx9) [inline configure.ac]: @tchaikov Don't think so? res_query and res_nquery have different number of arguments.

res_nquery(res_state statep, const char *dname, int class, int type, unsigned char *answer, int anslen)
res_query(const char *dname, int class, int type, unsigned char *answer, int anslen)

http://man7.org/linux/man-pages/man3/resolver.3.html

Comment (tchaikov) [inline configure.ac]: oh, sorry again. bad at counting! =(

**Comment by tchaikov** [comment]
lgtm

**Comment by tchaikov** [comment]
tested at 

* http://pulpito.ceph.com/kchai-2016-07-02_20:22:26-rados-wip-kefu-testing---basic-smithi/
* http://pulpito.ceph.com/kchai-2016-07-02_09:03:22-rados-wip-kefu-testing---basic-mira/

**Pull request merged**